sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryWriteConfiguration.java

```diff
@@ -0,0 +1,218 @@
+          "Invalid create disposition (%s) was specified. Available dispositions are: %s",
+          createDisposition,
+          createDispositions);
+    }
+    String writeDisposition = getWriteDisposition();
+    if (writeDisposition != null && !writeDisposition.isEmpty()) {
+      List<String> writeDispostions =
+          Arrays.stream(BigQueryIO.Write.WriteDisposition.values())
+              .map(w -> w.name())
+              .collect(Collectors.toList());
+      Preconditions.checkArgument(
+          writeDispostions.contains(writeDisposition.toUpperCase()),
+          "Invalid write disposition (%s) was specified. Available dispositions are: %s",
+          writeDisposition,
+          writeDispostions);
+    }
+
+    ErrorHandling errorHandling = getErrorHandling();
+    if (errorHandling != null) {
+      checkArgument(
+          !Strings.isNullOrEmpty(errorHandling.getOutput()),
+          invalidConfigMessage + "Output must not be empty if error handling specified.");
+    }
+
+    Boolean autoSharding = getAutoSharding();
+    Integer numStreams = getNumStreams();
+    if (autoSharding != null && autoSharding && numStreams != null) {
+      checkArgument(
+          numStreams == 0,
+          invalidConfigMessage
+              + "Cannot set a fixed number of streams when auto-sharding is enabled. Please pick only one of the two options.");
+    }
+  }
+
+  /** Instantiates a {@link BigQueryWriteConfiguration.Builder} instance. */
+  public static Builder builder() {
+    return new AutoValue_BigQueryWriteConfiguration.Builder();
+  }
+
+  @SchemaFieldDescription(
+      "The bigquery table to write to. Format: [${PROJECT}:]${DATASET}.${TABLE}")
+  public abstract String getTable();
+
+  @SchemaFieldDescription(
+      "Optional field that specifies whether the job is allowed to create new tables. "
+          + "The following values are supported: CREATE_IF_NEEDED (the job may create the table), CREATE_NEVER ("
+          + "the job must fail if the table does not exist already).")
+  @Nullable
+  public abstract String getCreateDisposition();
+
+  @SchemaFieldDescription(
+      "Specifies the action that occurs if the destination table already exists. "
+          + "The following values are supported: "
+          + "WRITE_TRUNCATE (overwrites the table data), "
+          + "WRITE_APPEND (append the data to the table), "
+          + "WRITE_EMPTY (job must fail if the table is not empty).")
+  @Nullable
+  public abstract String getWriteDisposition();
+
+  @SchemaFieldDescription(
+      "Determines how often to 'commit' progress into BigQuery. Default is every 5 seconds.")
+  @Nullable
+  public abstract Long getTriggeringFrequencySeconds();
+
+  @SchemaFieldDescription(
+      "This option enables lower latency for insertions to BigQuery but may ocassionally "
+          + "duplicate data elements.")
+  @Nullable
+  public abstract Boolean getUseAtLeastOnceSemantics();
```

Comment:
Seems like this is not usable since we do not expose the write method through this API and hence that will be explicitly set ?

Reply:
We need to this to keep backwards compatibility (this is exposed to the Python wrapper)

Reply:
Also when we do switch on at-least-once, we'll likely have to use this parameter